src/gadgets/WiretapWithEarplugs.hpp

```diff
@@ -0,0 +1,39 @@
+//
+// Created by jonas on 02.04.20.
+//
+
+#ifndef LIBCOMMON_WIRETAPWITHEARPLUGS_HPP
+#define LIBCOMMON_WIRETAPWITHEARPLUGS_HPP
+
+#include <Gadget.hpp>
+#include <util/UUID.hpp>
+#include <optional>
+
+namespace spy::gadget {
+    class WiretapWithEarplugs : public Gadget {
+    public:
+        WiretapWithEarplugs() : Gadget{GadgetEnum::WIRETAP_WITH_EARPLUGS}, working{true}, activeOn{std::nullopt} {};
+
+        [[nodiscard]] bool isWorking() const {
+            return working;
+        }
+
+        void setWorking(bool isWorking) {
+            WiretapWithEarplugs::working = isWorking;
+        }
+
+        [[nodiscard]] const std::optional<spy::util::UUID> &getActiveOn() const {
+            return activeOn;
+        }
+
+        void setActiveOn(const std::optional<spy::util::UUID> &active) {
+            WiretapWithEarplugs::activeOn = active;
+        }
```

Comment:
Ich finds ehrlich gesagt komisch, dass du da die ganze Definition mit in die .hpp packst, Wenn da dann später auch noch die zwei Methoden für die json (De-)Serialisierung dabei sind ist der header schon etwas aufgebläht. Was spricht dagegen die Definitionen jeweils in die .cpp zu verschieben?

Reply:
Mach ich. War mir egal und die default-option beim erstellen mit clion